packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -10,39 +10,135 @@ import 'framework.dart';
 }
 
-/// A widget that automatically does a [FadeTransition] between a new widget and
+/// Signature for builders used to generate custom transitions for
+/// [AnimatedChildSwitcher].
+///
+/// The [child] should be transitioned in when [animation] is running in the
+/// forward direction,
+///
+/// The function should return a widget which wraps the given [child]. It may
+/// also use the [animation] to inform its transition. It must not return null.
+typedef Widget AnimatedChildSwitcherTransitionBuilder(Widget child, Animation<double> animation);
+
+/// Signature for builders used to generate custom layouts for
+/// [AnimatedChildSwitcher].
+///
+/// The [children] should be positioned according to their corresponding
+/// animation.
+///
+/// The function should return a widget which contains the given children, laid
+/// out as desired. It must not return null.
+typedef Widget AnimatedChildSwitcherLayoutBuilder(List<Widget> children);
+
+/// A widget that by default does a [FadeTransition] between a new widget and
 /// the widget previously set on the [AnimatedChildSwitcher] as a child.
 ///
-/// More than one previous child can exist and be fading out while the newest
-/// one is fading in if they are swapped fast enough (i.e. before [duration]
-/// elapses).
+/// If they are swapped fast enough (i.e. before [duration] elapses), more than
+/// one previous child can exist and be transitioning out while the newest one
+/// is transitioning in.
+///
+/// If the "new" child is the same widget type as the "old" child, but with
+/// different parameters, then [AnimatedChildSwitcher] will *not* do a
+/// transition between them, since as far as the framework is concerned, they
+/// are the same widget, and the existing widget can be updated with the new
+/// parameters. If you wish to force the transition to occur, set a [Key]
+/// (typically a [UniqueKey] or [ValueKey], depending upon the desired behavior)
+/// on each child widget that you wish to be considered unique.
+///
+/// ## Sample code
+///
+/// ```dart
+/// class ClickCounter extends StatefulWidget {
+///   const ClickCounter({Key key}) : super(key: key);
+///
+///   @override
+///   _ClickCounterState createState() => new _ClickCounterState();
+/// }
+///
+/// class _ClickCounterState extends State<ClickCounter> {
+///   int count = 0;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return new Material(
+///       child: Column(
+///         mainAxisAlignment: MainAxisAlignment.center,
+///         children: <Widget>[
+///           new AnimatedChildSwitcher(
+///             duration: const Duration(milliseconds: 200),
+///             transitionBuilder: (Widget child, Animation<double> animation) {
+///               return new ScaleTransition(child: child, scale: animation);
+///             },
+///             child: new Text(
+///               '$count',
+///               key: new UniqueKey(), // Must have this to build a unique widget.
```

Comment:
This should be a `ValueKey<int>(count)`. Otherwise, rotating the phone (for instance) would cause the text to fade.

Reply:
OK.